sklearn/feature_extraction/hashing.py

```diff
@@ -155,11 +162,10 @@ def transform(self, raw_X):
             raw_X = (_iteritems(d) for d in raw_X)
         elif self.input_type == "string":
             raw_X = (((f, 1) for f in x) for x in raw_X)
-        indices, indptr, values = \
+        indices, indptr, values, self.feature_to_index_map_ = \
```

Comment:
We should only really be updating this when called in fit. or fit_transform. with `save_mappings=True`, fit would have to call fit_transform.

Reply:
But since FeatureHasher works even if we call the `.transform()` method without calling `.fit()` or `.fit_transform()`, this would not initialise `self.feature_to_index_map_` wouldn't that be inconsistent? I have done a `hasattr` check in the `get_feature_names` though.